library/VTKExtensions/Rendering/vtkF3DRendererWithColoring.cxx

```diff
@@ -804,9 +804,17 @@ std::string vtkF3DRendererWithColoring::ComponentToString(int component)
   }
   else
   {
-    std::string ret = "Component #";
-    ret += std::to_string(component);
-    return ret;
+    const char* compName = this->ArrayForColoring->GetComponentName(component);
+    if (compName == nullptr)
+    {
+      std::string ret = "Component #";
+      ret += std::to_string(component);
+      return ret;
+    } 
+    else 
+    {
+      return std::string(compName);
```

Comment:
could you add a data and test for this ? see cmake/testing.cmake

Reply:
Added 3 tests in cmake/testing.cmake